server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -359,6 +375,116 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
         return new SortFieldAndFormat(field, fieldType.docValueFormat(null, null));
     }
 
+    /**
+     * Return true if the primary sort in the provided <code>source</code>
+     * is an instance of {@link FieldSortBuilder}.
+     */
+    public static boolean hasPrimaryFieldSort(SearchSourceBuilder source) {
+        return getPrimaryFieldSortOrNull(source) != null;
+    }
+
+    /**
+     * Return the {@link FieldSortBuilder} if the primary sort in the provided <code>source</code>
+     * is an instance of this class, null otherwise.
+     */
+    public static FieldSortBuilder getPrimaryFieldSortOrNull(SearchSourceBuilder source) {
+        if (source == null || source.sorts() == null || source.sorts().isEmpty()) {
+            return null;
+        }
+        return source.sorts().get(0) instanceof FieldSortBuilder ? (FieldSortBuilder) source.sorts().get(0) : null;
+    }
+
+    /**
+     * Return a {@link Function} that converts a serialized point into a {@link Number} according to the provided
+     * {@link NumberFieldType}.
+     */
+    private static Function<byte[], Comparable> numericPointConverter(SortField sortField, NumberFieldType numberFieldType) {
+        Function<byte[], Number> converter = v -> numberFieldType.parsePoint(v);
+        switch (IndexSortConfig.getSortFieldType(sortField)) {
+            case LONG:
+                return v -> converter.apply(v).longValue();
+
+            case INT:
+                return v -> converter.apply(v).intValue();
+
+            case DOUBLE:
+                return v -> converter.apply(v).doubleValue();
+
+            case FLOAT:
+                return v -> converter.apply(v).floatValue();
```

Comment:
out of curiosity, why is it not ok to return just converter.apply(v) in the above cases?

Reply:
can you leave a comment about it?

Reply:
This is because we allow sort fields to force their numeric type through the `numeric_type` option:
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html#_sorting_numeric_fields
I left a comment in https://github.com/elastic/elasticsearch/commit/310b14025edb907c25ca6ede55c0eae556956a2f